query-engine/query-engine/src/server/mod.rs

```diff
@@ -0,0 +1,199 @@
+#![deny(missing_docs)]
+
+use super::dmmf;
+use crate::{
+    context::PrismaContext,
+    request_handlers::{
+        graphql::{GraphQLSchemaRenderer, GraphQlRequestHandler},
+        PrismaRequest, RequestHandler,
+    },
+    PrismaResult,
+};
+use elapsed_middleware::ElapsedMiddleware;
+
+use datamodel::{Configuration, Datamodel};
+use query_core::schema::QuerySchemaRenderer;
+use serde_json::json;
+use tide::http::{mime, StatusCode};
+use tide::{Body, Request, Response};
+
+use std::net::SocketAddr;
+use std::sync::Arc;
+
+mod elapsed_middleware;
+
+//// Shared application state.
+pub(crate) struct State {
+    cx: Arc<PrismaContext>,
+    enable_playground: bool,
+}
+
+impl State {
+    /// Create a new instance of `State`.
+    fn new(cx: PrismaContext, enable_playground: bool) -> Self {
+        Self {
+            cx: Arc::new(cx),
+            enable_playground,
+        }
+    }
+}
+
+impl Clone for State {
```

Comment:
Why explicit clone?

Reply:
The bound for `State` is `Send + Sync + Clone + 'static`. We were nesting `Arc`s before, while all we needed to do was clone the inner `Arc` and copy some data. Keeping `Clone` as the outer bound seems overall better.